PostPairsTrading.md

```diff
@@ -0,0 +1,323 @@
+
+![Spreads](https://github.com/hablapps/pairstrading/blob/5-Post/Spreads.gif?raw=true)
+
+
+## Two steps forward, one step back.
+
+Before proceeding to plot the NASDAQ100-SP500 spreads, we need to first plan our algorithm. In this post, we intend to create **a real-time scenario** for Pairs trading, so careful planning is essential for our sake.
+
+Decomposing our steps, let's start from the very beginning:
+
+1. Data management part
+
+        1.1. Read data
+
+        1.2. Filter data
+
+2. Outside the loop
+
+        2.1. Linear regression
+
+        2.2. Initialize values 
+
+3. Inside the loop
+
+        3.1. Calculate spreads
+
+        3.2. Write on buffers
+
+
+As previously mentioned, historical data is crucial for generating accurate spreads. We need to calculate each spread in real-time **using precomputed alpha and beta values** derived from both prices. Therefore, once we have obtained our historical values, we can proceed with calculating our linear regression.
+
+---
+
+```q
+// Fix data and take log(prices) -> Simulated data
+priceX: 0!1_(update delta:0f^deltas dateTime from 
+        distinct select distinct dateTime, log bid, log ask from
+        (update dateTime:"P"$@[;19;:;"."] each dateTime from 
+        tab1) where not null bid);
+
+priceY: 0!1_(update delta:0f^deltas dateTime from 
+        distinct select distinct dateTime, log bid, log ask from 
+        (update dateTime:"P"$@[;19;:;"."] each dateTime from 
+        tab2) where not null bid);
+
+// Read historical data
+historial_tab2: 1_ flip `open`high`low`close`adjClose`vol!("FFFFFF";",") 0: `:/data/stocks/NASDAQ100_hist.csv;
+
+historial_tab1: 1_ flip `open`high`low`close`adjClose`vol!("FFFFFF";",") 0: `:data/stocks/SP500_hist.csv;
+```
+
+---
+
+## Linear regression
+
+Now, armed with logarithms, we can replicate the process from our previous example and calculate both alpha and beta values. To do this, we'll start by taking the historical data for NASDAQ100 and SP500 and applying our linear regression function to derive the desired values.
+
+---
+
+```q
+// Calculate alpha and beta from historical values
+beta_lr: betaF[px:-100#log historial_tab1`close;py:-100#log historial_tab2`close]; // we only take most recent 100 values 
+
+alpha_lr: alphaF[px;py];
+```
+
+---
+
+Now we just calculate our spreads as we did before like:
+
+---
+
+```q
+spread: priceY[.streamPair.i][`bid] - ((priceX[.streamPair.i][`bid] * beta_lr)+alpha_lr);
+```
+
+---
+
+> 💡 You may notice that we retrieve bid price data from our price stream using an index (`.streamPair.i`). This is because we are simulating the arrival of these records dynamically, based on some delta time.

```

Comment:
Ok, now `.streamPair.i` is explained, but to be honest, I don't get this explanation.